src/Workspaces/Core/Portable/WorkspacesResources.resx

```diff
@@ -456,4 +456,7 @@
   <data name="Exceptions" xml:space="preserve">
     <value>Exceptions:</value>
   </data>
+  <data name="ImmutableArrayIncorrectlyInitialized" xml:space="preserve">
+    <value>ImmutableArray was incorrectly initialized, consider using ImmutableArray&lt;T&gt;.Empty</value>
```

Comment:
Don't you throw on 'Empty' as well? Shouldn't this be more like "DiagnosticIds cannot be null or empty"?